proxy/proxy/client.go

```diff
@@ -150,10 +150,19 @@ func (c *Client) handleConn(conn Conn) {
 		conn.Conn = dbgConn{conn.Conn}
 	}
 
-	c.Conns.Add(conn.Instance, conn.Conn)
-	copyThenClose(server, conn.Conn, conn.Instance, "local connection on "+conn.Conn.LocalAddr().String())
+	entry := &ConnSetEntry{conn.Conn, time.Now()}
```

Comment:
I think a less intrusive way to implement this would be to create a wrapped around `conn.Conn` that overrides the `Write` and `Read` to track the last time written. 

With a wrapper I think you should be able to get await without changing the interface in several places? It would also be more efficient without the need for another go-routine/channel (you could probably just use a mutex to guard the values instead) .